lavalink/node.py

```diff
@@ -502,12 +502,91 @@ def register_state_handler(self, func):
     def unregister_state_handler(self, func):
         self._state_handlers.remove(func)
 
-    async def join_voice_channel(self, guild_id, channel_id, deafen: bool = False):
+    async def create_player(self, channel: VoiceChannel, deafen: bool = False) -> Player:
+        """
+        Connects to a discord voice channel.
+
+        This function is safe to repeatedly call as it will return an existing
+        player if there is one.
+
+        Parameters
+        ----------
+        channel
+
+        Returns
+        -------
+        Player
+            The created Player object.
         """
-        Alternative way to join a voice channel if node is known.
+        if self._already_in_guild(channel):
+            p = self.get_player(channel.guild.id)
+            await p.move_to(channel, deafen=deafen)
+        else:
+            p: Player = await channel.connect(cls=Player)  # type: ignore
+            if deafen:
+                await p.guild.change_voice_state(channel=p.channel, self_deaf=True)
+            self._players_dict[channel.guild.id] = p
+            await self.refresh_player_state(p)
```

Comment:
This seems to already be done by `Player.connect()` which is called by d.py above.